File: Sorting/bubbleSortRecursive.js
function bubbleSort(arr, k, i, isSwapped){
    if(k===0) return ;
    if(i===k){
        if(isSwapped===false) return;
        bubbleSort(arr, k-1, 0, false);
        return;
    }
    if(arr[i]>arr[i+1]) {
        let temp = arr[i];
        arr[i] = arr[i+1];
        arr[i+1] = temp;
        isSwapped = true;
    }
    bubbleSort(arr, k, i+1, isSwapped);
}

let arr = [9, 0, 3, 8, 7, 5];
bubbleSort(arr, arr.length-1, 0, false);
console.log(arr);

File: Sorting/bubbleSortRecursive.cpp
#include <iostream>
using namespace std;

void bubbleSort(int arr[], int k, int i, bool isSwapped){
    if(k==0) return ;
    if(i==k){
        if(isSwapped==false) return;
        bubbleSort(arr, k-1, 0, false);
        return;
    }
    if(arr[i]>arr[i+1]) {
        swap(arr[i], arr[i+1]);
        isSwapped=true;
    }
    bubbleSort(arr, k, i+1, isSwapped);
}

int main() {
    int n=7;
    int arr[] = {6, 8, 10, 15, -5, 0, 4};
    bubbleSort(arr, n-1, 0, false);
    for(int i=0; i<n; i++) cout<<arr[i]<<" ";
    return 0;
}